contrib/hack/start-server.sh

```diff
@@ -23,20 +23,36 @@ export PATH=${ROOT}/contrib/hack:${PATH}
 # Clean up old containers if still around
 docker rm -f etcd apiserver > /dev/null 2>&1 || true
 
-# Start etcd, our DB
+# Start etcd, our DB. Also, map 6443 (api server port) to some random
+# port on the host - then ask Docker for the port # as we'll need to use
+# that when we talk to it.
 echo Starting etcd
-docker run -ti --name etcd -d --net host quay.io/coreos/etcd > /dev/null
+docker run -ti --name etcd -d -p 6443 quay.io/coreos/etcd > /dev/null
+PORT=$(docker port etcd 6443 | sed "s/.*://")
 
 # And now our API Server
 echo Starting the API Server
-docker run -ti --name apiserver -d --net host \
+docker run -tid --name apiserver \
 	-v ${ROOT}:/go/src/github.com/kubernetes-incubator/service-catalog \
```

Comment:
@pmorie aside from the ":z" what other change are you proposing?  adding `--privileged` ?